### ISSUE TITLE
Add email notifications

### ISSUE THREAD

**Issue by jliliamen** [tellform/tellform#223]
Tellform would be complete with email notifications. Thanks!

**Comment by itaditya** [comment]
I would like to work on this

**Comment by marykatefain** [comment]
I could also use this! I want to use Tellform on a project but email notifications are a must. Thank you!

**Comment by whitef0x0** [comment]
@itaditya Let me know if you need any help getting started

**Comment by alexgleason** [comment]
It looks like this project is *already* configured to send out email. It's just a matter of calling nodemailer's mail function in the right place and with a relevant email message. In fact, I see that happening already [here](https://github.com/tellform/tellform/blob/e6b9f83494c2c004a5c7253d86b8baf92bf6c8f2/app/controllers/users/users.password.server.controller.js#L84) for account confirmations, so we could apply that same method in the spot where the form is saved.

Based on skimming the code, it seems we need to put the mailer within [`app/controllers/forms.server.controller.js` under `submission.save()`](https://github.com/tellform/tellform/blob/4414b0e3903c30f7bdc6cd383cb4493e36df87a9/app/controllers/forms.server.controller.js#L69).

This actually seems trivial, like something we could do in 30 minutes with testing.

My only real concern is that we will need per-form settings to make this really useful to most people.  That increases the complexity a bit. Eg, Mary is a developer at an agency who creates a contact form. When the contact form is submitted, it should notify 3 people in the sales department but not Mary. This is a common scenario so we'll need to account for it.

It looks like new settings should go under [`app/models/form.server.model.js` in the `FormSchema` object](https://github.com/tellform/tellform/blob/4414b0e3903c30f7bdc6cd383cb4493e36df87a9/app/models/form.server.model.js#L82). I don't know Mongoose, so I don't know which types to use for these settings.

I think we'll need:

* `emailNotifications` *(Boolean)* - enable or disable email notifications when this form is submitted (default: false)
* `emailRecipients` *(String)* - emails of people who should be notified when this form is submitted. For multiple recipients, separate them with commas. (default: empty string)

I'm not really sure how to pull these in on the front-end dashboard or the right types to use for Mongoose since I'm unfamiliar with Mongoose.

**Comment by whitef0x0** [comment]
@jliliamen I can help you with the front-end and mongoose stuff if you want to give it a shot.

**Comment by whitef0x0** [comment]
@jliliamen @itaditya @marykatefain @alexgleason @dlackty I have created a PR for this feature https://github.com/tellform/tellform/pull/230.

From what I understand, what is meant by email notifications is a simple email that is send whenever a form is submitted?

Can you give me an example of how this email should be formatted and should it contain the submission values as well?

**Comment by jliliamen** [comment]
What I meant by email notification is that tellform should send the creator of the form an email telling that a user has used an email.  Perhaps something like this
Subject: New TellForm Contact (Email of the person who used the form)
Body: Answers in the form. I don't have any preference here.

**Comment by marykatefain** [comment]
I agree with @jliliamen, that's exactly the same thing I'm looking for. Thanks!

**Comment by whitef0x0** [comment]
@jliliamen is it ok if the email of the person who used the form is only in the body and not in the subject?

**Comment by whitef0x0** [comment]
@marykatefain

**Comment by marykatefain** [comment]
@whitef0x0 yeah that would be fine for me!

**Comment by whitef0x0** [comment]
Just to update everyone, this is currently added to the 2.20 release that is scheduled to be pushed to production in the next week or two.

**Comment by whitef0x0** [comment]
@jliliamen @itaditya @marykatefain @alexgleason @dlackty Email notifications are now live at http://admin.tellform.com!

**Comment by jliliamen** [comment]
@whitef0x0 : you're the man. Thanks!

**Comment by jliliamen** [comment]
@whitef0x0 : i cannot add variables in self notifications the combobox is empty

**Comment by whitef0x0** [comment]
@jliliamen I'm actually going to revert these changes. A few bugs I've missed.
